src/deferred.js

```diff
@@ -157,6 +157,10 @@ jQuery.extend( {
 											mightThrow();
 										} catch ( e ) {
 
+											if ( jQuery.Deferred.exceptionHook ) {
+												jQuery.Deferred.exceptionHook( e );
```

Comment:
If we're comfortable with the above, though, a hook here seems like an interesting approach. Since only the exception is passed, there's no access to a relevant thenable and therefore no value for anything other than debugging.